Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -1192,5 +1266,108 @@ extension NIOAsyncWriter {
                 preconditionFailure("Invalid state")
             }
         }
+
+        /// Actions returned by `sinkFinish()`.
+        @usableFromInline
+        enum UnbufferQueuedEventsAction {
+            case callDidYield(Delegate, Deque<Element>)
+            case callDidYieldElement(Delegate, Element)
+            case callDidTerminate(Delegate, Error?)
+        }
+
+        @inlinable
+        /* fileprivate */ internal mutating func unbufferQueuedEvents() -> UnbufferQueuedEventsAction? {
+            switch self._state {
+            case .initial:
+                preconditionFailure("Invalid state")
+
+            case .streaming(let isWritable, let inDelegateOutcall, let cancelledYields, let suspendedYields, var elements, let delegate):
+                precondition(inDelegateOutcall, "We must be in a delegate outcall when we unbuffer events")
+
+                if elements.count == 0 {
+                    // Nothing to do. We haven't gotten any writes.
+                    self._state = .streaming(
+                        isWritable: isWritable,
+                        inDelegateOutcall: false, // We can now indicate that we are done with the outcall
+                        cancelledYields: cancelledYields,
+                        suspendedYields: suspendedYields,
+                        elements: elements,
+                        delegate: delegate
+                    )
+                    return .none
+                } else if elements.count >= 1 {
+                    // We have to yield all of the elements now.
+                    self._state = .streaming(
+                        isWritable: isWritable,
+                        inDelegateOutcall: inDelegateOutcall,
+                        cancelledYields: cancelledYields,
+                        suspendedYields: suspendedYields,
+                        elements: .init(),
+                        delegate: delegate
+                    )
+
+                    return .callDidYield(delegate, elements)
+
+                } else {
```

Comment:
You'll never hit this branch because the condition above is `>= 1` but should be `> 1`

Reply:
Thanks. Good catch!